SpatialGDK/Source/Private/Interop/GlobalStateManager.cpp

```diff
@@ -188,3 +256,137 @@ bool UGlobalStateManager::IsSingletonEntity(Worker_EntityId EntityId)
+
+			// Set the AcceptingPlayers state on the GSM
+			Schema_AddBool(UpdateObject, SpatialConstants::GLOBAL_STATE_MANAGER_ACCEPTING_PLAYERS_ID, uint8_t(bInAcceptingPlayers));
+
+			// Component updates are short circuited so we set the updated state here and then send the component update.
+			bAcceptingPlayers = bInAcceptingPlayers;
+			NetDriver->Connection->SendComponentUpdate(GlobalStateManagerEntityId, &Update);
+		}
+	});
+
+	Receiver->AddEntityQueryDelegate(RequestID, GSMQueryDelegate);
+}
+
+void UGlobalStateManager::AuthorityChanged(bool bWorkerAuthority, Worker_EntityId CurrentEntityID)
+{
+	// Make sure the GameInstance knows that this worker is now authoritative over the GSM (used for server travel).
+	// The GameInstance is the only persistent object during server travel.
+	// bIsWorkerAuthorativeOverGSM exists to inform the worker that was previously authoritative over the GSM that it has the responsibility of loading the new snapshot.
+	Cast<USpatialGameInstance>(NetDriver->GetWorld()->GetGameInstance())->bIsWorkerAuthorativeOverGSM = bWorkerAuthority;
+
+	// Also update this instance of the GSM that it has current authority (used for accepting players toggle).
+	// The instance of each GSM is destroyed on server travel and a new one is made, hence the need for 'live' authority.
+	bHasLiveMapAuthority = bWorkerAuthority;
+
+	// Make sure we update our known entity id for the GSM when we receive authority.
+	GlobalStateManagerEntityId = CurrentEntityID;
+
+	OnAuthorityChanged.ExecuteIfBound(bWorkerAuthority);
+}
+
+// Queries for the GlobalStateManager in the deployment.
+// bWithRetry will continue querying until the state of AcceptingPlayers is true, this is so clients know when to connect to the deployment.
+void UGlobalStateManager::QueryGSM(bool bWithRetry)
+{
+	Worker_ComponentConstraint GSMComponentConstraint{};
+	GSMComponentConstraint.component_id = SpatialConstants::GLOBAL_STATE_MANAGER_MAP_URL;
+
+	Worker_Constraint GSMConstraint;
+	GSMConstraint.constraint_type = WORKER_CONSTRAINT_TYPE_COMPONENT;
+	GSMConstraint.component_constraint = GSMComponentConstraint;
+
+	Worker_EntityQuery GSMQuery{};
+	GSMQuery.constraint = GSMConstraint;
+	GSMQuery.result_type = WORKER_RESULT_TYPE_SNAPSHOT;
+
+	Worker_RequestId RequestID;
+	RequestID = NetDriver->Connection->SendEntityQueryRequest(&GSMQuery);
+
+	EntityQueryDelegate GSMQueryDelegate;
+	GSMQueryDelegate.BindLambda([this, bWithRetry](Worker_EntityQueryResponseOp& Op)
+	{
+		if (Op.status_code != WORKER_STATUS_CODE_SUCCESS || Op.result_count == 0)
+		{
+			UE_LOG(LogGlobalStateManager, Error, TEXT("Could not find GSM via entity query: %s"), UTF8_TO_TCHAR(Op.message));
+			if (bWithRetry)
+			{
+				UE_LOG(LogGlobalStateManager, Warning, TEXT("Retrying entity query for the GSM in 3 seconds"));
+				FTimerHandle RetryTimer;
+				TimerManager->SetTimer(RetryTimer, [this, bWithRetry]()
+				{
+					QueryGSM(bWithRetry);
+				}, 3.f, false);
+			}
+		}
+
+		if (Op.result_count == 1)
```

Comment:
else if?

Reply:
or return after the previous and check that result_count == 1

Reply:
Checkf added.